decidim-core/app/controllers/concerns/decidim/needs_permission.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require "active_support/concern"
+
+module Decidim
+  # Common logic to work with the permissions system
+  module NeedsPermission
+    extend ActiveSupport::Concern
+
+    included do
+      helper_method :allowed_to?
+
+      class ::Decidim::MissingPermission < StandardError
+      end
+
+      rescue_from Decidim::MissingPermission, with: :user_not_authorized
+
+      def check_permission_to(action, subject, extra_context = {})
+        raise Decidim::MissingPermission unless allowed_to?(action, subject, extra_context)
+      end
+
+      def allowed_to?(action, subject, extra_context = {})
+        permission_class.new(
+          current_user,
+          { scope: permission_scope.to_s, action: action.to_s, subject: subject.to_s },
```

Comment:
You have the opportunity to build a `PermissionsAction` object as I suggested up above.

Reply:
Done!